src/main/java/no/finn/unleash/repository/FeatureToggleRepository.java

```diff
@@ -41,13 +58,18 @@ private Runnable updateToggles() {
         return () -> {
             try {
                 FeatureToggleResponse response = toggleFetcher.fetchToggles();
-                if (response.getStatus() == FeatureToggleResponse.Status.CHANGED) {
+                eventDispatcher.dispatch(response);
+                FeatureToggleResponse.Status status = response.getStatus();
+                if (status == FeatureToggleResponse.Status.CHANGED) {
                     toggleCollection = response.getToggleCollection();
                     toggleBackupHandler.write(response.getToggleCollection());
-                } else {
-                    LOG.warn("Error fetching toggles from Unleash API (StatusCode: {})", response.getHttpStatusCode());
+                } else if (status == FeatureToggleResponse.Status.UNAVAILABLE){
+                    String errorMessage = "Error fetching toggles from Unleash API - StatusCode: " + response.getHttpStatusCode();
+                    eventDispatcher.dispatch(new UnleashException(errorMessage, null));
```

Comment:
Here we might end up dispatching twice for same event, both in line 61 and here.

Reply:
My thinking here was that this case is interesting for subscribers reacting to onError(), instead of having to check the status of FeatureToggleResponse. Alternatively, onError() could be called conditionally from FeatureToggleResponse.publishTo() if that's better. Or I could remove it altogether. What do you prefer?

Reply:
I see your point, but it might be the modelling here that is a bit off. 

I don't think that the user cares if it is a error on the HTTP layer or if it some kind of IO-expection. Either there is a response (success) or it's not (error). Is there a use-case where the user would like to get a `FeatureToggleResponse` if it was not successful (instead of an error)? 


I would prefer that `onError` was called conditionally from `FeatureToggleResponse.publishTo()`.

Reply:
If the client receives 4xx- or 5xx-responses from the unleash server or some proxy server, the status of FeatureToggleResponse will be `UNAVAILABLE` and no exception will be thrown. The simplest way I can think of to achieve this is to take down the unleash server behind a loadbalancer which then respons with 404s or 503s. 

We had an incident in production where a misbehaving client managed to overwhelm the unleash server (actually the postgres database). Since we had tied alerts to the `UNAVAILABLE`-status, the problem was highlighted quickly to us in what turned out to be almost a pleasing experience, displaying the self-healing properties of both unleash and our infrastructure. 

I'll refactor the `else if`, taking your comments below about logging into account!